### PR TITLE
Changing walkthrough design

### PR DESCRIPTION
**Pull request by agusanchez**
## Useful links
- [Contributing](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CONTRIBUTING.md) has more information and tips for a great pull request. This will give you a nice introduction about how to contribute with Andes.
- [Wiki](https://github.com/mercadolibre/fury_andesui-android/wiki) has all the info you need to start with Andes. Even I heard there's a video there...

## Motivation
Tell us why you are here and why you took time to develop new code for our beloved lib.

## Description
Please describe your intentions with this new code you want to introduce inside Andes UI lib.
Try to answer those questions:
- What is it about?
- How does it work?

## Affected component
Is highly probable that this new code directly affects one (or more?) components inside this lib. Tell us who they are!

## Frontify component link
The UX team behind Andes keeps all the Andes Components inside the Frontify site. Each component has its own section. Let us know the section of the component you want to modify.

## Screenshots
This is a UI library, delight us with some stunning images.

Screenshot_1611691218
Screenshot_1611691235
Screenshot_1611691243
Screenshot_1611691248
Screenshot_1611691252

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [ ] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [ ] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

File: coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/CoachmarkContainerView.kt [hunk 1]
```diff
@@ -24,15 +24,12 @@ class CoachmarkContainerView @JvmOverloads constructor(
         listener = l
     }
 
-    fun setData(buttonText: String, position: Int, size: Int) {
-        nextButton.text = buttonText
-        nextButton.setOnClickListener { listener?.onClickNextButton(position) }
+    fun setData(position: Int, size: Int) {
```
Comment (eportasML) [inline coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/CoachmarkContainerView.kt:27]: Estamos haciendo un breaking change aca. Por que?

File: coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/walkthroughmessage/WalkthroughMessageView.kt [hunk 2]
```diff
@@ -18,18 +23,50 @@ class WalkthroughMessageView @JvmOverloads constructor(
...
         walkthroughDescription.typeface = context.getFontOrDefault(R.font.andes_font_regular)
+        walkthroughNextButton.typeface = context.getFontOrDefault(R.font.andes_font_semibold)
     }
 
-    fun setData(data: WalkthroughMessageModel) {
-        walkthroughTitle.text = data.title
-        walkthroughDescription.text = data.description
-        walkthroughTitle.visibility = View.VISIBLE
-        walkthroughDescription.visibility = View.VISIBLE
+    fun setListener(l: WalkthroughScrollessMessageView.WalkthroughButtonClicklistener) {
```
Comment (cvillamilmeli) [inline coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/walkthroughmessage/WalkthroughMessageView.kt:35]: Don't use generic names for variables
```suggestion
    fun setListener(walkthroughButtonClicklistener: WalkthroughScrollessMessageView.WalkthroughButtonClicklistener) {
```

File: coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/walkthroughmessage/WalkthroughMessageView.kt [hunk 3]
```diff
@@ -18,18 +23,50 @@ class WalkthroughMessageView @JvmOverloads constructor(
 ) : ConstraintLayout(context, attrs, defStyleAttr) {
 
     private var position: WalkthroughMessagePosition = WalkthroughMessagePosition.BELOW
+    private var listener: WalkthroughScrollessMessageView.WalkthroughButtonClicklistener? = null
```
Comment (cvillamilmeli) [inline coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/walkthroughmessage/WalkthroughMessageView.kt:26]: Don't use generic names for variables
```suggestion
    private var walkthroughButtonClicklistener: WalkthroughScrollessMessageView.WalkthroughButtonClicklistener? = null
```

File: coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/walkthroughmessage/WalkthroughMessageView.kt [hunk 4]
```diff
@@ -18,18 +23,50 @@ class WalkthroughMessageView @JvmOverloads constructor(
...
+        } else {
+            walkthroughNextButton.setBackgroundResource(R.drawable.andes_walkthrough_configuration_button_background)
+        }
+
+        if (data.title.isNotEmpty()) {
+            walkthroughTitle.text = data.title
+            walkthroughTitle.visibility = View.VISIBLE
+        } else {
+            walkthroughTitle.visibility = View.GONE
+        }
```
Comment (cvillamilmeli) [inline coachmark/src/main/java/com/mercadolibre/android/andesui/coachmark/view/walkthroughmessage/WalkthroughMessageView.kt:55]: You can use [this function](https://github.com/mercadolibre/fury_andesui-android/pull/229/commits/39548e394d0ddddd40307b7a10b1f17a822b2070#diff-3f94a57502df05dc38f0e067adfdd773e2cd93bbcbedc9088ada343aca22fbafR132) also for this case

File: coachmark/src/main/res/layout/andes_walkthrough_message.xml [hunk 5]
```diff
@@ -54,17 +54,37 @@
...
             app:layout_constraintTop_toBottomOf="@+id/walkthroughTitle"
             tools:text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor."
             tools:visibility="visible" />
 
+        <TextView
+            android:id="@+id/walkthroughNextButton"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="32dp"
+            android:textSize="16sp"
```
Comment (cvillamilmeli) [inline coachmark/src/main/res/layout/andes_walkthrough_message.xml:68]: can you use dimens please

**Review by cvillamilmeli — CHANGES_REQUESTED** [review]
Some changes to review

**Comment by agusanchez** [comment]
@cvillamilmeli Ya quedaron los cambios en el PR.

File: coachmark/src/main/res/layout/andes_walkthrough_container.xml [hunk 6]
```diff
@@ -42,10 +35,10 @@
         android:id="@+id/counterText"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginStart="24dp"
         android:textColor="@color/andes_white"
         android:textSize="14sp"
```
Comment (jperezgonzal) [inline coachmark/src/main/res/layout/andes_walkthrough_container.xml:39]: Pasar esto a dimens

File: coachmark/src/main/res/layout/andes_walkthrough_container.xml [hunk 7]
```diff
@@ -55,47 +48,20 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginEnd="24dp"
```
Comment (jperezgonzal) [inline coachmark/src/main/res/layout/andes_walkthrough_container.xml:50]: Pasar esto a dimens

File: coachmark/src/main/res/layout/andes_walkthrough_container.xml [hunk 8]
```diff
@@ -20,13 +20,6 @@
         android:orientation="horizontal"
         app:layout_constraintGuide_begin="80dp" />
```
Comment (jperezgonzal) [inline coachmark/src/main/res/layout/andes_walkthrough_container.xml:21]: Pasar esto a dimens

**Review by cvillamilmeli — APPROVED** [review]

**Pull request merged**